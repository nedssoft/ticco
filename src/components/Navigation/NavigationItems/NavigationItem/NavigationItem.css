.NavigationItem {
  display: block;
  width: 100%;
}

.NavigationItem a {
  color: black;
  text-decoration: none;
  display: block;
  padding: 10px;
  font-size: 24px;
}

.NavigationItem a:hover,
.NavigationItem a:active,
.NavigationItem a.active {
  color: aliceblue;
  background: #032B43;
}

@media (min-width: 500px) {
  .NavigationItem {
    margin: 0;
    display: flex;
    height: 100%;
    width: auto;
    align-items: center;
  }
  
  .NavigationItem a {
    color: aliceblue;
    text-decoration: none;
    padding: 20px;
    border-bottom: 4px solid transparent;
    display: block;
  }
  
  .NavigationItem a:hover,
  .NavigationItem a:active,
  .NavigationItem a.active {
    background: #032B43;
    border-bottom: 6px solid aliceblue;
    color: aliceblue;
  }
}