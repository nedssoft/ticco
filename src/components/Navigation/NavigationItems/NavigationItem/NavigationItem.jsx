import React from 'react'
import { NavLink } from 'react-router-dom'
import classes from './NavigationItem.css';

function NavigationItem  ({exact, link, children }){
  
  return (
    <li className={classes.NavigationItem}>
      <NavLink 
        to={link}
        activeClassName={classes.active}
        exact={exact}
      >
        {children}

      </NavLink>
    </li>
  )
}

export default NavigationItem
