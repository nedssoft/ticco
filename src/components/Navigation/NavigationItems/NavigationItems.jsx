import React from 'react'
import classes from './NavigationItems.css';
import NavigationItem from './NavigationItem/NavigationItem';

const NavigationItems = () => {
  return (
    <ul className={classes.NavigationItems}>
      <NavigationItem link="/orders">Sign In</NavigationItem> 
      <NavigationItem link="/auth">Sign Up</NavigationItem>
      <NavigationItem link="/logout">Logout</NavigationItem>
    </ul>
  )
}

export default NavigationItems
