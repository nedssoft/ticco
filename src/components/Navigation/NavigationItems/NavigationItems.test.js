import React from 'react';
import { shallow } from 'enzyme'
import NavigationItems from './NavigationItems'
import NavigationItem from './NavigationItem/NavigationItem'

describe('<NavigationItems', () => {
  it('should render three navigation items', ()=> {
    const component = shallow(
      <NavigationItems />
    );
    expect(component).toMatchSnapshot();
    expect(component.find(NavigationItem)).toHaveLength(3)
  })
})