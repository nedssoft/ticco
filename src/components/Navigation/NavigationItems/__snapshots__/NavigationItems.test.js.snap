// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItems should render three navigation items 1`] = `
<ul
  className="NavigationItems"
>
  <NavigationItem
    link="/signin"
  >
    Sign In
  </NavigationItem>
  <NavigationItem
    link="/signup"
  >
    Sign Up
  </NavigationItem>
  <NavigationItem
    link="/logout"
  >
    Logout
  </NavigationItem>
</ul>
`;
